centinel/experiments/baseline.py

```diff
@@ -0,0 +1,255 @@
+                    http_netloc = urlparse_object.netloc
+
+                domain_name = http_netloc.split(':')[0]
+
+                http_path = urlparse_object.path
+                if http_path == '':
+                    http_path = '/'
+
+                # we assume scheme is either empty, or "http", or "https"
+                # other schemes (e.g. "ftp") are out of the scope of this
+                # measuremnt
+                if urlparse_object.scheme == "https":
+                    http_ssl = True
+                    if len(http_netloc.split(':')) == 2:
+                        ssl_port = http_netloc.split(':')[1]
+
+            except Exception as exp:
+                logging.warning("%s: failed to parse URL: %s" % (url, exp))
+                http_netloc = url
+                http_ssl    = False
+                ssl_port = 443
+                http_path   = '/'
+                domain_name = url
+
+            # start tcpdump
+            td = Tcpdump()
+            tcpdump_started = False
+
+            try:
+                if self.record_pcaps:
+                    td.start()
+                    tcpdump_started = True
+                    logging.info("%s: tcpdump started..." % (url))
+                    # wait for tcpdump to initialize
+                    time.sleep(1)
+            except Exception as exp:
+                logging.warning("%s: tcpdump failed: %s" % (url, exp))
+
+            # HTTP GET
+            logging.info("%s: HTTP" % (url))
+            try:
+                http_results[url] = http.get_request(http_netloc,
+                                                     http_path,
+                                                     ssl=http_ssl)
+            except Exception as exp:
+                logging.info("%s: HTTP test failed: %s" %
+                             (url, exp))
+                http_results[url] = { "exception" : str(exp) }
+
+            # TLS certificate
+            # this will only work if the URL starts with https://
+            if http_ssl:
+                try:
+                    tls_result = {}
+                    logging.info("%s: TLS certificate" %
+                                 (domain_name))
+                    fingerprint, cert = tls.get_fingerprint(domain_name, ssl_port)
+                    tls_result['port'] = ssl_port
+                    tls_result['fingerprint'] = fingerprint
+                    tls_result['cert'] = cert
+
+                    tls_results[domain_name] = tls_result
+                except Exception as exp:
+                    logging.info("%s: TLS certfiticate download failed: %s" %
+                                 (domain_name, exp))
+                    tls_results[domain_name] = { "exception" : str(exp) }
+
+            # DNS Lookup
+            logging.info("%s: DNS" % (domain_name))
+            try:
+                dns_results[domain_name] = dnslib.lookup_domain(domain_name)
```

Comment:
> I am ok with merging the rest of the baseline stuff now, then refactoring with another pull request as long as we change the timeout value.

Alright. I'll restore the timeout value and push the rest in another pull request.